numpyro/distributions/transforms.py

```diff
@@ -168,18 +152,28 @@ def codomain(self):
         else:
             raise NotImplementedError
 
-    @property
-    def event_dim(self):
-        return 1 if self.domain is constraints.real_vector else 0
-
     def __call__(self, x):
         return self.loc + self.scale * x
 
     def _inverse(self, y):
         return (y - self.loc) / self.scale
 
     def log_abs_det_jacobian(self, x, y, intermediates=None):
-        return sum_rightmost(jnp.broadcast_to(jnp.log(jnp.abs(self.scale)), jnp.shape(x)), self.event_dim)
+        return jnp.broadcast_to(jnp.log(jnp.abs(self.scale)), jnp.shape(x))
+
+
+def _get_compose_transform_input_event_dim(parts):
+    input_event_dim = parts[-1].domain.event_dim
+    for part in parts[len(parts) - 1::-1]:
+        input_event_dim = part.domain.event_dim + max(input_event_dim - part.codomain.event_dim, 0)
+    return input_event_dim
+
+
+def _get_compose_transform_output_event_dim(parts):
+    output_event_dim = parts[0].codomain.event_dim
+    for part in parts[1:]:
+        output_event_dim = part.codomain.event_dim + max(output_event_dim - part.domain.event_dim, 0)
+    return output_event_dim
```

Comment:
This is a bit trickier in case any parts change event dim. I believe there are correct implementations in https://github.com/pytorch/pytorch/pull/50581/files#diff-9c38272c807be20a5832ca58bc3b1f3454c70d30e28ecbd39ca40191219f7d81

Reply:
I just saw that we are making duplicated effort (they are equivalent, I did comment about the reason in your PR). ;)